inst/tests/tests.Rraw

```diff
@@ -18702,3 +18686,120 @@ test(2234.7, DT[, min(.SD), by=c(.I)], data.table(I=1L:5L, V1=c(1L, 2L, 3L, 2L,
 test(2234.8, DT[, min(.SD), by=.I%%2L], error="by.*contains .I.*supported")  # would be nice to support in future; i.e. by odd/even rows, and by=(.I+1L)%/%2L for pairs of rows; i.e. any expression of .I
 test(2234.9, DT[, min(.SD), by=somefun(.I)], error="by.*contains .I.*supported")
 
+# add na.rm argument to first/last, #4446 #4239
+DT = data.table( grp = INT(c(1,1,     2,2,2,               3,          4,4)),
+                 A =   INT(c(1,NA,    NA,2,NA,             NA,         3,3)),
+                 B =       c(NA,1,    pi,3,NA,             3,          pi,NA),
+                 C =       c("a",NA,  NA,"b",NA,           "c",        "d",NA),
+                 D =       c(NA,1i,   2i,3i,NA,            NA,         4i,NA),
+                 E =    list(NA,NA,   NA,c("a","b"),NULL,  list(1:2),  3:4,NA))
+# to encourage adding cases to DT above (some NAs, all NAs, NAs beginning and/or end within group, types, etc) we
+# define helper functions FNNA and LNNA here to construct the correct result to test against. Explicitly coding
+# all the results instead may reveal mistakes and be good but then might be inflexible to adding more cases.
+FNNA = function(x) {  # First Not NA
+  nna = which(!is.na(x) & !(is.list(x) & sapply(x,is.null)))
```

Comment:
is this the same as `nna = which(!( if (is.list(x)) sapply(x, is.null) else is.na(x))`? <- seems easier to follow to me

Reply:
not the same but changed to the following agree easier: `nna = which(if (is.list(x)) !is.na(x) & !vapply_1b(x,is.null) else !is.na(x))`

Reply:
perfect thanks